experimental/Schemes/Resolution_structure.jl

```diff
@@ -241,6 +380,45 @@ function _desing_curve(X::AbsCoveredScheme, I_sl::AbsIdealSheaf)
   return phi
 end
 
+function _desing_lipman(X::AbsCoveredScheme, I_sl::AbsIdealSheaf)
+  dim(X) == 2 || error("Lipman's algorithm is not applicable")
+
+  if dim(I_sl) == 1
+    Xnorm, phi = normalization(X)
+    incs = phi.inclusions
+    f = initialize_mixed_blow_up_sequence(phi,I_sl)
+    f_initialized = true
+  end    
+  
+  I_sl_temp = ideal_sheaf_of_singular_locus(Xnorm)
+  decomp = maximal_associated_points(I_sl_temp)
+
+  while !is_one(I_sl_temp)
+    while length(decomp) > 0
+      I = small_generating_set(pop!(decomp))
```

Comment:
In my opinion `small_generating_set` does not make sense for ideal sheaves. We should take care that `simplify` does what is needed here and use that.

Reply:
Currently (while WIP), I use small_generating_set locally as a testing ground -- without messing up simplify and all the instances where it is applied. In the end, this should, of course, be a simplify (possibly with something changed in a suitable way).

Reply:
I will leave this in for now. It will be changed in a future PR, as soon as the desingularizations and principalizations have made more progress.